### PR TITLE
[PkgConfigDeps] Duplicated names order

### PR DESCRIPTION
**Pull request by franramirez688**
Docs for: https://github.com/conan-io/conan/pull/10344 https://github.com/conan-io/conan/pull/10263

### CODE REVIEW

File: reference/conanfile/tools/gnu/pkgconfigdeps.rst [hunk 1]
```diff
@@ -181,6 +181,29 @@ Their ``pkg_config_name`` properties are used as the final PC file names:
     Requires: zlib
 
 
+A special mention when a component shares the same final PC name as the root package one:
```
Comment (czoido) [inline reference/conanfile/tools/gnu/pkgconfigdeps.rst:184]: Is PC a frequent way to refer to pkg-config ? I think it's better to directly say `pkg-config`

Comment (franramirez688) [reply reference/conanfile/tools/gnu/pkgconfigdeps.rst:184]: I'm referring to any `****.pc` file

Comment (memsharded) [reply reference/conanfile/tools/gnu/pkgconfigdeps.rst:184]: Call it "``.pc`` file name" for clarity maybe

File: reference/conanfile/tools/gnu/pkgconfigdeps.rst [hunk 2]
```diff
@@ -181,6 +181,29 @@ Their ``pkg_config_name`` properties are used as the final PC file names:
...
+            self.cpp_info.set_property("pkg_config_name", "OpenCL")  # -> OpenCL.pc
+            self.cpp_info.components["_opencl-headers"].set_property("pkg_config_name", "OpenCL")  # -> OpenCL.pc
+
+The only ``*.pc`` file created will be the one belonging to the component:
+
+- OpenCL.pc (from component)
+
+.. important::
+
+    Components ``*.pc`` file names have more priority than package one in case of collision.
```
Comment (czoido) [inline reference/conanfile/tools/gnu/pkgconfigdeps.rst:204]: I don't know if this important note is worth it...

Comment (franramirez688) [reply reference/conanfile/tools/gnu/pkgconfigdeps.rst:204]: I agree. I was doubting if putting it or not. I guess it's enough with the explanation before.

**Review by memsharded — APPROVED** [review]

**Pull request merged**